pkg/openstackconfigversion/git_util.go

```diff
@@ -228,3 +228,12 @@ func SyncGit(
 
 	return configVersions, nil
 }
+
+// truncateDiff  truncate the diff size to less than 2MB
+func truncateDiff(diff string, log logr.Logger) string {
+	if len(diff) > 2097151 {
```

Comment:
two things:
* the max message size is 2MB, so we need to be lower than that for other object data fields
* while the max massage send size is 2MB, the default max size we can store in etcd is 1MB. maybe we should just truncate to 512KB.